bundles/framework/myplaces3/ButtonHandler.js

```diff
@@ -307,45 +301,43 @@ Oskari.clazz.define("Oskari.mapframework.bundle.myplaces3.ButtonHandler",
              * @param {Oskari.mapping.drawtools.event.DrawingEvent} event
              */
             'DrawingEvent': function (event) {
-                if (event.getId() !== this.instance.getName() || this.instance.isEditPlace()){
+                if (event.getId() !== this.instance.getName() || this.instance.isEditPlace()) {
                     return;
                 }
-                if (event.getIsFinished()){
-                    if (!this.instance.isFinishedDrawing()){
-                        this.featureAdded(event.getData());
-                    } else {
-                        //handle Finished drawing -> main view
-                        // set ignore so we don't cancel our drawing unintentionally
-                        this.ignoreEvents = true;
-                        // ask toolbar to select default tool
-                        var toolbarRequest = Oskari.requestBuilder('Toolbar.SelectToolButtonRequest')();
-                        this.instance.sandbox.request(this, toolbarRequest);
-                        // disable ignore to act normally after ^request
-                        this.ignoreEvents = false;
-                        if (this.dialog) {
-                            this.dialog.close();
-                        }
-                    }
-                } else {
+                if (!event.getIsFinished()) {
                     this.activeDrawing(event.getData());
+                    return;
+                }
+                if (!this.instance.isFinishedDrawing()) {
+                    this.featureAdded(event.getData());
+                    return;
+                }
+                // handle Finished drawing -> main view
+                // set ignore so we don't cancel our drawing unintentionally
+                this.ignoreEvents = true;
+                // ask toolbar to select default tool
+                this.instance.sandbox.request(this, Oskari.requestBuilder('Toolbar.SelectToolButtonRequest')());
+                // disable ignore to act normally after ^request
+                this.ignoreEvents = false;
```

Comment:
This assumes requests are synchronous so it might easily break. I think sandbox.request() might be sync, but postRequestByName is async.